k2/python/k2/fsa.py

```diff
@@ -226,14 +224,14 @@ def __setattr__(self, name: str, value: Any) -> None:
         '''
 
         if hasattr(type(self), name) or name in self.__dict__:
-            # For attribute names that have a getter/setter function (like 'property'
-            # or which exist as members of the object's __dict__ (like
-            # 'arcs', '_tensor_attr', '_non_tensor_attr'), do the member lookup
-            # in the normal way as if __setattr__ was not defined.
-            object.__setattr__(key, value)
-
-        if name in ('_tensor_attr', '_non_tensor_attr', 'arcs',
-                    '_cache'):
+            # For attribute names that have a getter/setter function
+            # (like 'property' or which exist as members of the object's
+            # __dict__ (like 'arcs', '_tensor_attr', '_non_tensor_attr'),
+            # do the member lookup in the normal way as if __setattr__ was
+            # not defined.
+            object.__setattr__(name, value)
+
+        if name in ('_tensor_attr', '_non_tensor_attr', 'arcs', '_cache'):
```

Comment:
To avoid having to do this we can, in __init__ _init_internal, have it do things like, instead of setting self.arcs, self._tensor_attr, etc., self.__dict__['arcs'] = arcs and so on.